### PR TITLE
Return an error if we request more than 1Mb of memory in copy_address_raw

### PR DESCRIPTION
**Pull request by jvns**
there's an issue (#64) where if some memory is incorrect, we can end up passing in an invalid length to `copy_address_raw` (length = some huge number), then allocating huge amounts of memory in `copy_address_raw` and the program crashes. This PR doesn't fix the underlying issue (why is the memory in an invalid state?), but it should avoid allocating huge amounts of memory and instead log an error and continue. The choice of 1MB is arbitrary.

In the future we should be more disciplined about logging more detailed output than this when there's an error (so it's possible to ask users for the debug output)

### CODE REVIEW

**Comment by liaden** [comment]
Is this a case of the callstack being too deep?

**Comment by jvns** [comment]
I don't know what the root cause is! I'm going to make another PR that prints more debug output when there's an error.

File: src/copy.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@ pub enum MemoryCopyError {
     #[fail(display = "Failed to copy memory address {:x}", _0)] Io(usize, #[cause] std::io::Error),
     #[fail(display = "Process isn't running")] ProcessEnded,
     #[fail(display = "Other")] Other,
+    #[fail(display = "Too much memory requested when copying")] InvalidLength,
```
Comment (kamalmarhubi) [inline src/copy.rs]: I'd make the name be `RequestTooLarge` or something more indicative of the actual issue. Also, take a `usize` parameter and include it in the display message.

File: src/copy.rs [hunk 2]
```diff
@@ -36,6 +37,9 @@ where
     T: CopyAddress,
 {
     debug!("copy_address_raw: addr: {:x}", addr as usize);
+    if length > 1000000 {
```
Comment (kamalmarhubi) [inline src/copy.rs]: make this a constant up top?

**Pull request merged**